apps/sms/test/marionette/mocks/mock_navigator_moz_mobile_message.js

```diff
@@ -32,6 +32,70 @@ Services.obs.addObserver(function(document) {
     }
   }
 
+  function generateCanvas(width, height) {
+    var canvas = document.createElement('canvas');
+    var context = canvas.getContext('2d');
+
+    canvas.width = width;
+    canvas.height = height;
+
+    var linearGradient = context.createLinearGradient(0, 0, width, height);
+    linearGradient.addColorStop(0, 'blue');
+    linearGradient.addColorStop(1, 'red');
+
+    context.fillStyle = linearGradient;
+    context.fillRect (0, 0, width, height);
+
+    return canvas;
+  }
+
+  function generateImageBlob(width, height, type, quality) {
+    var canvas = generateCanvas(width, height);
+
+    return new Promise((resolve) => {
+      canvas.toBlob((blob) => {
+        canvas.width = canvas.height = 0;
+        canvas = null;
+
+        resolve(blob);
+      }, type, quality);
+    });
+  }
+
+  function rehydrateAttachments(storage) {
+    var attachmentPromises = [];
```

Comment:
Somewhat ugly method, haven't found significantly better way to rewrite it yet - so keeping as is....